packages/babel-core/test/config-chain.js

```diff
@@ -49,8 +64,13 @@ function loadOptions(opts) {
   return babel.loadOptions({ cwd: __dirname, ...opts });
 }
 
-function loadOptionsAsync(opts) {
-  return babel.loadOptionsAsync({ cwd: __dirname, ...opts });
+function loadOptionsAsync({ filename, cwd = __dirname }, mjs) {
+  if (mjs) {
+    // import() crashes with jest
+    return loadOptionsAsyncInSpawedProcess({ filename, cwd });
```

Comment:
@SimenB This is needed to spawn tests using `import()` in a different process so that Jest doesn't have to handle them.

If you remove this line (and thus run the tests in the current Node.js process managed by Jest), and run `node --experimental-vm-modules ./node_modules/.bin/jest babel-core/test/config-chain` it crashes.

The file containing `import()` is https://github.com/babel/babel/blob/main/packages/babel-core/src/config/files/import.js, and you can Ctrl+F for `"babel.config.mjs"` in this file to find a test using `import()` internally.

I'll now create a branch with a smaller example.

Reply:
yeah, this is weird. The `importModuleDynamically` callback isn't invoked at all - I'm guessing this is some edge case in Node's implementation

Reply:
I think it will be better if I post the reproduction to the Node.js repo then: even if the tests or Jest did something wrong (but from my superficial debugging I don't think so), Node.js shouldn't crash.

Reply:
From debugging - putting a breakpoint on the `import()` expression shows it is called with an `URL` with the following `href`: `file:///var/folders/gj/0mygpdfn6598xh34njlyrqzc0000gn/T/babel-test-load-config-async-babel.config.mjsYKe5KD/babel.config.mjs`. This file exists in disk and is valid ESM. Then doing "Step over next function call" crashes with a segfault without stopping the debugger

Reply:
Ref: https://github.com/nodejs/node/issues/35889